### ISSUE TITLE
Service selector update is ignored when handle update event

### ISSUE THREAD

**Issue by lminzhw** [kubernetes/kubernetes#108923]
### What happened?

I met this issue when I create a LoadBalancer type service with wrong selector. And my LoadBalancer and cloud-controller-manager did nothing even after I correct my service selector.

I tried to find what happend inside my cloud-controller-manager, and finally I got this func    https://github.com/kubernetes/kubernetes/blob/8dd52e6c5dcd8068ee1080a4b7dcabd0a140a5ee/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L506

This func handle the update event by checking many LoadBalancer related attributes, but the service selector is not checked.

Is my analysis correct? Is that a feature or a bug?

Look forward to your reply.

### What did you expect to happen?

My LoadBalancer will be able to receive the selector change event, and do something after selecto change.

### How can we reproduce it (as minimally and precisely as possible)?

A LoadBalancer type service with wrong selector is enough.

### Kubernetes version

<details>

```console
$ kubectl version
1.19.16
```

The version of cloud-controller-manager is my inner version based on 1.19.16. But as I know, the func `needsUpdate` is the same in branch master.

</details>

### Cloud provider

<details>
a k8s.io/cloud-provider issue?
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by lzhecheng** [comment]
@lminzhw hello, what you described looks similar to the issue I solved recently but I am not sure they're the same because 1) the issue couldn't be triggered easily 2) it shouldn't be related to `needsUpdate()`. I suggest trying to reproduce with master code (my change hasn't been in any release branch).

**Comment by bridgetkromhout** [comment]
Hi, @lminzhw - as @lzhecheng explained, this looks like the issue which he resolved with https://github.com/kubernetes/kubernetes/pull/107631. That patch is now available in the latest releases of 1.24, 1.23, 1.22, and 1.21. I see that you were using Kubernetes 1.19.16 when you reported the bug. That version is unsupported, so we won't be backporting the fix. 

Please try one of the latest releases of Kubernetes: 1.21.13, 1.22.10, 1.23.7, 1.24.1 - and let us know if you can reproduce the problem with one of these. Thanks!
